pkg/reconciler/revision/resources/pa.go

```diff
@@ -36,8 +37,16 @@ func MakePA(rev *v1alpha1.Revision) *av1alpha1.PodAutoscaler {
 			Namespace: rev.Namespace,
 			Labels:    makeLabels(rev),
 			Annotations: resources.FilterMap(rev.GetAnnotations(), func(k string) bool {
-				// Ignore last pinned annotation.
-				return k == serving.RevisionLastPinnedAnnotationKey
+				switch k {
+				case serving.RevisionLastPinnedAnnotationKey:
+					// Ignore last pinned annotation.
+					return true
+				case autoscaling.MinScaleAnnotationKey:
```

Comment:
I'd rather see the KPA be aware of the last pinned annotation than have the Revision be aware of this autoscaling annotation.

Reply:
Sounds good, I'll move it to the KPA

Reply:
> I'd rather see the KPA be aware of the last pinned annotation than have the Revision be aware of this autoscaling annotation.

Curious why? Revisions already "know" about autoscalers because it's responsible for creating them -- I don't love having the KPA "know" about revisions as well.

Reply:
Annotations are not part of the spec, so sensitivity to them is a layering violation.

Reply:
Put differently, if I implement my own autoscaler, I don't need to respect that :)